### ISSUE TITLE
Llama.cpp with IPEX-LLM is crashing

### ISSUE THREAD

**Issue by kimtaeyo** [intel-analytics/ipex-llm#11711]
I followed the instruction here to install Core Ultra 9 MTL 185H : https://github.com/intel-analytics/ipex-llm/blob/main/docs/mddocs/Quickstart/llama_cpp_quickstart.md

It worked 3-4 months ago but not right now and showing like this error message. I tried 155H and 165H but error messages are all same. ARC driver 32.0.101.5768 and OneAPI is 2024.1

----

(llm-cpp) PS C:\Users\kimtaeyo\PRJ\llama-cpp> .\main -m .\models\mistral-7b-instruct-v0.1.Q4_K_M.gguf -n 32 --prompt "Once upon a time, there existed a little girl who liked to have adventures. She wanted to go to places and meet new people, and have fun" -t 8 -e -ngl 33 --color -sm none
Log start
main: build = 1 (b791c1a)
main: built with IntelLLVM 2024.0.2 for
main: seed  = 1722811827
llama_model_loader: loaded meta data with 20 key-value pairs and 291 tensors from .\models\mistral-7b-instruct-v0.1.Q4_K_M.gguf (version GGUF V2)
llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.
llama_model_loader: - kv   0:                       general.architecture str              = llama
llama_model_loader: - kv   1:                               general.name str              = mistralai_mistral-7b-instruct-v0.1
llama_model_loader: - kv   2:                       llama.context_length u32              = 32768
llama_model_loader: - kv   3:                     llama.embedding_length u32              = 4096
llama_model_loader: - kv   4:                          llama.block_count u32              = 32
llama_model_loader: - kv   5:                  llama.feed_forward_length u32              = 14336
llama_model_loader: - kv   6:                 llama.rope.dimension_count u32              = 128
llama_model_loader: - kv   7:                 llama.attention.head_count u32              = 32
llama_model_loader: - kv   8:              llama.attention.head_count_kv u32              = 8
llama_model_loader: - kv   9:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010
llama_model_loader: - kv  10:                       llama.rope.freq_base f32              = 10000.000000
llama_model_loader: - kv  11:                          general.file_type u32              = 15
llama_model_loader: - kv  12:                       tokenizer.ggml.model str              = llama
llama_model_loader: - kv  13:                      tokenizer.ggml.tokens arr[str,32000]   = ["<unk>", "<s>", "</s>", "<0x00>", "<...
llama_model_loader: - kv  14:                      tokenizer.ggml.scores arr[f32,32000]   = [0.000000, 0.000000, 0.000000, 0.0000...
llama_model_loader: - kv  15:                  tokenizer.ggml.token_type arr[i32,32000]   = [2, 3, 3, 6, 6, 6, 6, 6, 6, 6, 6, 6, ...
llama_model_loader: - kv  16:                tokenizer.ggml.bos_token_id u32              = 1
llama_model_loader: - kv  17:                tokenizer.ggml.eos_token_id u32              = 2
llama_model_loader: - kv  18:            tokenizer.ggml.unknown_token_id u32              = 0
llama_model_loader: - kv  19:               general.quantization_version u32              = 2
llama_model_loader: - type  f32:   65 tensors
llama_model_loader: - type q4_K:  193 tensors
llama_model_loader: - type q6_K:   33 tensors
llm_load_vocab: special tokens definition check successful ( 259/32000 ).
llm_load_print_meta: format           = GGUF V2
llm_load_print_meta: arch             = llama
llm_load_print_meta: vocab type       = SPM
llm_load_print_meta: n_vocab          = 32000
llm_load_print_meta: n_merges         = 0
llm_load_print_meta: n_ctx_train      = 32768
llm_load_print_meta: n_embd           = 4096
llm_load_print_meta: n_head           = 32
llm_load_print_meta: n_head_kv        = 8
llm_load_print_meta: n_layer          = 32
llm_load_print_meta: n_rot            = 128
llm_load_print_meta: n_swa            = 0
llm_load_print_meta: n_embd_head_k    = 128
llm_load_print_meta: n_embd_head_v    = 128
llm_load_print_meta: n_gqa            = 4
llm_load_print_meta: n_embd_k_gqa     = 1024
llm_load_print_meta: n_embd_v_gqa     = 1024
llm_load_print_meta: f_norm_eps       = 0.0e+00
llm_load_print_meta: f_norm_rms_eps   = 1.0e-05
llm_load_print_meta: f_clamp_kqv      = 0.0e+00
llm_load_print_meta: f_max_alibi_bias = 0.0e+00
llm_load_print_meta: f_logit_scale    = 0.0e+00
llm_load_print_meta: n_ff             = 14336
llm_load_print_meta: n_expert         = 0
llm_load_print_meta: n_expert_used    = 0
llm_load_print_meta: causal attn      = 1
llm_load_print_meta: pooling type     = 0
llm_load_print_meta: rope type        = 0
llm_load_print_meta: rope scaling     = linear
llm_load_print_meta: freq_base_train  = 10000.0
llm_load_print_meta: freq_scale_train = 1
llm_load_print_meta: n_yarn_orig_ctx  = 32768
llm_load_print_meta: rope_finetuned   = unknown
llm_load_print_meta: ssm_d_conv       = 0
llm_load_print_meta: ssm_d_inner      = 0
llm_load_print_meta: ssm_d_state      = 0
llm_load_print_meta: ssm_dt_rank      = 0
llm_load_print_meta: model type       = 7B
llm_load_print_meta: model ftype      = Q4_K - Medium
llm_load_print_meta: model params     = 7.24 B
llm_load_print_meta: model size       = 4.07 GiB (4.83 BPW)
llm_load_print_meta: general.name     = mistralai_mistral-7b-instruct-v0.1
llm_load_print_meta: BOS token        = 1 '<s>'
llm_load_print_meta: EOS token        = 2 '</s>'
llm_load_print_meta: UNK token        = 0 '<unk>'
llm_load_print_meta: LF token         = 13 '<0x0A>'
[SYCL] call ggml_init_sycl
ggml_init_sycl: GGML_SYCL_DEBUG: 0
ggml_init_sycl: GGML_SYCL_F16: no
Native API failed. Native API returns: -30 (PI_ERROR_INVALID_VALUE) -30 (PI_ERROR_INVALID_VALUE)
Exception caught at file:C:/Users/Administrator/actions-runner/cpp-release/_work/llm.cpp/llm.cpp/llama-cpp-bigdl/ggml-sycl.cpp, line:13140, func:operator()
ggml_backend_sycl_set_single_device: use single device: [0]
GGML_ASSERT: C:/Users/Administrator/actions-runner/cpp-release/_work/llm.cpp/llm.cpp/llama-cpp-bigdl/ggml-sycl.cpp:18283: main_gpu_id<g_all_sycl_device_count

**Comment by rnwang04** [comment]
Hi @kimtaeyo ,
This error seems cause by llama.cpp can't find sycl device .
Could you please provide us more details with [this script](https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/scripts/env-check.sh) ?

**Comment by kimtaeyo** [comment]
Python 3.10.14
python: can't open file 'C:\\Users\\kimtaeyo\\Downloads\\check.py': [Errno 2] No such file or directory
-----------------------------------------------------------------
System Information

Host Name:                 MSI
OS Name:                   Microsoft Windows 11 Pro
OS Version:                10.0.22631 N/A Build 22631
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Intel
Registered Organization:
Product ID:                00355-61224-34719-AAOEM
Original Install Date:     4/30/2024, 1:26:35 PM
System Boot Time:          8/4/2024, 9:33:04 PM
System Manufacturer:       Micro-Star International Co., Ltd.
System Model:              Stealth 16 AI Studio A1VGG
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 170 Stepping 4 GenuineIntel ~2300 Mhz
BIOS Version:              American Megatrends International, LLC. E15F4IMS.010, 11/10/2023
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-05:00) Eastern Time (US & Canada)
Total Physical Memory:     32,430 MB
Available Physical Memory: 23,095 MB
Virtual Memory: Max Size:  34,478 MB
Virtual Memory: Available: 24,672 MB
Virtual Memory: In Use:    9,806 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\MSI
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB5039895
                           [02]: KB5027397
                           [03]: KB5040527
                           [04]: KB5040568
Network Card(s):           3 NIC(s) Installed.
                           [01]: Killer E3100G 2.5 Gigabit Ethernet Controller
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [02]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [03]: Realtek USB GbE Family Controller
                                 Connection Name: Ethernet 3
                                 DHCP Enabled:    Yes
                                 DHCP Server:     10.22.224.196
                                 IP address(es)
                                 [01]: 10.23.208.79
                                 [02]: fe80::b392:c9d6:f2bb:5e26
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
-----------------------------------------------------------------
'xpu-smi' is not recognized as an internal or external command,
operable program or batch file.
xpu-smi is not installed properly.

**Comment by rnwang04** [comment]
After offline sync with @kimtaeyo , it looks more like GPU driver issue.
We can't reproduce this error on our MTL machines. He will try same script on our MTL machine.
btw, could you please run [env-check.bat](https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/scripts/env-check.bat) of https://github.com/intel-analytics/ipex-llm/tree/main/python/llm/scripts to check system environment? The above information does not seem to be complete.

**Comment by kimtaeyo** [comment]
(llm-cpp) PS C:\Users\kimtaeyo\Downloads> .\env-check.bat
Python 3.11.9
-----------------------------------------------------------------
C:\Users\kimtaeyo\miniforge3\envs\llm-cpp\Lib\site-packages\transformers\utils\generic.py:441: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
transformers=4.36.2
-----------------------------------------------------------------
torch=2.2.0+cpu
-----------------------------------------------------------------
Name: ipex-llm
Version: 2.1.0b20240804
Summary: Large Language Model Develop Toolkit
Home-page: https://github.com/intel-analytics/ipex-llm
Author: BigDL Authors
Author-email: bigdl-user-group@googlegroups.com
License: Apache License, Version 2.0
Location: C:\Users\kimtaeyo\miniforge3\envs\llm-cpp\Lib\site-packages
Requires:
Required-by: llama-index-llms-ipex-llm
-----------------------------------------------------------------
C:\Users\kimtaeyo\miniforge3\envs\llm-cpp\Lib\site-packages\torchvision\io\image.py:13: UserWarning: Failed to load image Python extension: 'Could not find module 'C:\Users\kimtaeyo\miniforge3\envs\llm-cpp\Lib\site-packages\torchvision\image.pyd' (or one of its dependencies). Try using the full path with constructor syntax.'If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
IPEX is not installed properly.
-----------------------------------------------------------------
Total Memory: 31.670 GB

Chip 0 Memory: 16 GB | Speed: 5600 MHz
Chip 1 Memory: 16 GB | Speed: 5600 MHz
-----------------------------------------------------------------
CPU Manufacturer: GenuineIntel
CPU MaxClockSpeed: 2500
CPU Name: Intel(R) Core(TM) Ultra 9 185H
CPU NumberOfCores: 16
CPU NumberOfLogicalProcessors: 22
-----------------------------------------------------------------
GPU 0: Intel(R) Arc(TM) Graphics         Driver Version:  32.0.101.5768
-----------------------------------------------------------------
-----------------------------------------------------------------
System Information

Host Name:                 MSI
OS Name:                   Microsoft Windows 11 Pro
OS Version:                10.0.22631 N/A Build 22631
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Intel
Registered Organization:
Product ID:                00355-61224-34719-AAOEM
Original Install Date:     4/30/2024, 1:26:35 PM
System Boot Time:          8/4/2024, 9:33:04 PM
System Manufacturer:       Micro-Star International Co., Ltd.
System Model:              Stealth 16 AI Studio A1VGG
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 170 Stepping 4 GenuineIntel ~2300 Mhz
BIOS Version:              American Megatrends International, LLC. E15F4IMS.010, 11/10/2023
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-05:00) Eastern Time (US & Canada)
Total Physical Memory:     32,430 MB
Available Physical Memory: 22,413 MB
Virtual Memory: Max Size:  34,478 MB
Virtual Memory: Available: 23,879 MB
Virtual Memory: In Use:    10,599 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\MSI
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB5039895
                           [02]: KB5027397
                           [03]: KB5040527
                           [04]: KB5040568
Network Card(s):           3 NIC(s) Installed.
                           [01]: Killer E3100G 2.5 Gigabit Ethernet Controller
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [02]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [03]: Realtek USB GbE Family Controller
                                 Connection Name: Ethernet 3
                                 DHCP Enabled:    Yes
                                 DHCP Server:     10.22.224.196
                                 IP address(es)
                                 [01]: 10.23.208.79
                                 [02]: fe80::b392:c9d6:f2bb:5e26
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
-----------------------------------------------------------------
'xpu-smi' is not recognized as an internal or external command,
operable program or batch file.
xpu-smi is not installed properly.

**Comment by kimtaeyo** [comment]
The issue was resolved by eliminating a peculiar compatibility pack from Microsoft. Essentially, the installation of the ARC driver led to a forced installation of the OpenCL, OpenGL, and Vulkan Compatibility Pack by Microsoft. This action inadvertently blocked the system from locating Sycl devices. It’s advisable to escalate this issue as it appears to be prevalent in many MSI laptops.

![image](https://github.com/user-attachments/assets/0119cc1d-9cbc-47cb-934f-a10c0acc9def)

**Comment by rnwang04** [comment]
Hi @kimtaeyo , thanks for pointing this issue and providing us with this solution ! We will add this issue and the solution into our troubleshooting.

**Comment by rnwang04** [comment]
This issue is added here : https://github.com/intel-analytics/ipex-llm/blob/main/docs/mddocs/Quickstart/llama_cpp_quickstart.md#why-my-program-cant-find-sycl-device
